TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -62,42 +63,43 @@ class MetAPI{
             parameters: parameters) { result in
                 switch result {
                 case .failure:
-                    completion(nil)
+                    completion(.failure(.invalidUrlComponents))
                 case .success(let data):
                     let jsonDecoder = JSONDecoder()
                     jsonDecoder.keyDecodingStrategy = .useDefaultKeys
                     do {
                         let result = try jsonDecoder.decode(ObjectsResponse.self, from: data)
-                        completion(result)
+                        completion(.success(result))
                     } catch {
-                        completion(nil)
+                        completion(.failure(.noDataInResponse))
                     }
                 }
             }
     }
     
-    func object(id: ArtID, completion: @escaping (ObjectResponse?) -> Void) {
-        self.metAPICache.object(id: id) { [weak self] objectResponce in
+    func object(id: ArtID, completion: @escaping (Result<ObjectResponse, MetAPIError>) -> Void) {
+        self.metAPICache.object(id: id) { [weak self] objectResponse in
             guard let self = self else {
-                completion(nil)
+                completion(.failure(.metAPIDoesNotExist))
                 return
             }
-            if let objectResponce = objectResponce {
-                completion(objectResponce)
+            if let objectResponse = objectResponse {
+                completion(.success(objectResponse))
             } else {
-                self.executeObject(id: id) {[weak self] realResponse in
-                    guard let realResponse = realResponse else {
-                        completion(nil)
-                        return
+                self.executeObject(id: id) {[weak self] realResponseResult in
+                    switch realResponseResult {
+                    case .failure:
+                        completion(.failure(.noDataInResponse))
```

Comment:
This is not a `noDataInResponse`, use the error associated with the `failure` case:
```
case .failure(let error):
```